device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/MultiplexingClient.java

```diff
@@ -127,14 +133,44 @@ else if (receivePeriod == 0) //default builder value for this option, signals th
      * <p>
      * If this client is already open, then this method will do nothing.
      * <p>
-     * @throws IOException If any IO errors occur while opening the multiplexed connection.
+     * @throws MultiplexingClientException If any IO or authentication errors occur while opening the multiplexed connection.
+     * @throws MultiplexingClientDeviceRegistrationFailedException If one or many of the registered devices failed to authenticate.
+     * Any devices not found in the map of registration exceptions provided by this exception have registered successfully.
+     * Even when this is thrown, the AMQPS/AMQPS_WS connection is still open, and other clients may be registered to it.
      */
-    public void open() throws IOException
+    public void open() throws MultiplexingClientException
     {
         synchronized (this.operationLock)
         {
             log.info("Opening multiplexing client");
-            this.deviceIO.open();
+            try
+            {
+                this.deviceIO.openWithoutWrappingException();
+            }
+            catch (TransportException e)
```

Comment:
correct me if I am wrong, but `TransportException` is not in the `MultiplexingClientException` hierarchy 
From what I can tell, `MultiplexingClientException` extends Exception and `MultiplexingClientDeviceRegistrationFailedException` extends `MultiplexingClientException` so line 157 would not be true ever ! no?

Reply:
Line 157 is checking the nested exception of this caught exception. I did add one special case where the lower layer throws a ```TransportException``` with a nested ```MultiplexingClientDeviceRegistrationAuthenticationException``` so that we could handle this code path. In particular, it is when you call open after registering your devices.

Reply:
ah, I totallly missed the fact that you are checking the inner exception